term/src/traits.rs

```diff
@@ -0,0 +1,102 @@
+//! Abstractions for using terms and the various sub-types as trait-objects.
+//!
+//! _Note:_ This is a PoC implementation. Therefore, it covers only IRIs,
+//! literals and blank nodes. If this concept gets accepted the implementation
+//! should also cover variables.
+
+use super::*;
+
+/// Interface of RDF terms.
+pub trait Term {
+    /// Downcast to an IRI.
+    fn as_iri(&self) -> Option<&dyn Iri> {
+        None
+    }
+    /// Downcast to a literal.
+    fn as_literal(&self) -> Option<&dyn Literal> {
+        None
+    }
+    /// Downcast to a blank node.
+    fn as_blank_node(&self) -> Option<&dyn BlankNode> {
+        None
+    }
+}
+
+impl<'a> PartialEq for &'a dyn Term {
```

Comment:
Darn, I always forget we can do this (implementing a trait for a *reference* type). Nice.